### DISCUSSION TITLE
Spring boot web starter version

### DISCUSSION THREAD

**Discussion by rmishra3261** [Netflix/dgs-framework#558] [Q&A] [answered]
I need to use DGS library with Springboot web 2.1.x versions but it does not work and gives below error:

Parameter 0 of constructor in com.netflix.graphql.dgs.webmvc.autoconfigure.DgsWebMvcConfigurationProperties required a bean of type 'java.lang.String' that could not be found.

However, when I use the 2.3.9.RELEASE it works fine. Can anyone confirm, if DGS can be used with Springboot web 2.1.x.

Thanks.

**Comment by hantsy** [comment]
I think Spring Boot 2.4+ is ok. Not sure why you are stick on the old versions.

**Reply by rmishra3261** [reply]
Thanks @hantsy, there are existing apps on older Springboot versions that need to get data from the Graphql endpoint and hence the need.

**Reply by hantsy** [reply]
For those projects that are still active, adding a Github action depdendabot to automate the dependencies upgrade is easy.

**Comment by paulbakker** [comment, marked as answer]
@rmishra3261 We currently test with `2.3.9` and above. 
The reason we don't test with older versions is that it's simply too much work to maintain. The supported versions range from what we actively use at Netflix.

I would highly recommend keeping your Spring versions updated, but I understand that's not always in your control.

You do mention that those older applications are just getting data from a GraphQL endpoint. Are you only using the client? In that case, make sure to just use the `graphql-dgs-client` dependency, and not the whole starter. That way you might work around the problem.

**Reply by rmishra3261** [reply]
Thanks a lot @paulbakker, this worked. I also had to do a messy workaround with jackson-databind to first exclude the older version that came with spring boot 2.1.0 and then explicitly include the correct version.
